odl/contrib/fom/test/test_supervised.py

```diff
@@ -60,23 +94,135 @@ def test_mean_absolute_error(space):
     true = odl.phantom.white_noise(space)
     data = odl.phantom.white_noise(space)
 
-    result = odl.contrib.fom.mean_absolute_error(data, true)
+    result = fom.mean_absolute_error(data, true)
     expected = np.mean(np.abs(true - data))
 
     assert result == pytest.approx(expected)
 
 
 def test_psnr(space):
+    """Test the ``psnr`` fom."""
+
     true = odl.phantom.white_noise(space)
     data = odl.phantom.white_noise(space)
+    zero = space.zero()
 
-    result = odl.contrib.fom.psnr(data, true)
+    # Check the corner cases
+    assert fom.psnr(true, true) == np.inf
+    assert fom.psnr(zero, zero) == np.inf
+    assert fom.psnr(data, zero) == -np.inf
 
+    # Compute the true value
     mse = np.mean((true - data) ** 2)
     maxi = np.max(np.abs(true))
     expected = 10 * np.log10(maxi ** 2 / mse)
 
-    assert result == pytest.approx(expected)
+    # Test regular call
+    result = fom.psnr(data, true)
+    assert result == pytest.approx(expected, abs=1e-6)
+
+    # Test with force_lower_is_better giving negative of expected
+    result = fom.psnr(data, true, force_lower_is_better=True)
+    assert result == pytest.approx(-expected, abs=1e-6)
+
+    # Test with Z-score that result is independent of affine transformation
+    result = fom.psnr(data * 3.7 + 1.234, true, use_zscore=True)
+    expected = fom.psnr(data, true, use_zscore=True)
+    assert result == pytest.approx(expected, abs=1e-5)
+
+
+def test_ssim(space):
+    ground_truth = odl.phantom.white_noise(space)
+
+    # SSIM of true image should be either
```

Comment:
SSIM of true image compared to ?

Reply:
`SSIM of an image compared to itself should be either`